main/dapps/server/asset/index.ts

```diff
@@ -58,19 +59,17 @@ export default {
   dapp: async (res: ServerResponse, namehash: string) => { // Resolve dapp via IPFS, inject functionality and send it back to the client
     // if (!ipfs return error(res, 404, 'IPFS client not running')
     const cid = store('main.dapps', namehash, 'content')
-    const index = await nebula.ipfs.getFile(`${cid}/index.html`)
-    // res.setHeader('Set-Cookie', [`__app=${app}`, `__session=${session}`])
-    res.setHeader('Access-Control-Allow-Origin', '*')
-    res.setHeader('Access-Control-Allow-Headers', 'X-Requested-With,content-type')
-    res.writeHead(200)
-    const $ = cheerio.load(index)
-    // $('html').prepend(`
-    //   <script>
-    //     // const initial = ${JSON.stringify(storage.get(cid) || {})}
-    //     ${inject}
-    //   </script>
-    // `)
-    res.end($.html())
+
+    try { 
+      const index = await nebula.ipfs.getFile(`${cid}/index.html`)    
+      res.setHeader('Access-Control-Allow-Origin', '*')
+      res.setHeader('Access-Control-Allow-Headers', 'X-Requested-With,content-type')
+      res.writeHead(200)
+      const $ = cheerio.load(index)
+      res.end($.html())
+    } catch (e) {
+      log.error('could not resolve dapp', (e as NodeJS.ErrnoException).message)
```

Comment:
what happens in the UI once we catch this, does it just keep spinning forever?

Reply:
probably best is to display the fact that it cant be loaded, but this is better than crashing so maybe create a ticket for that functionality and merge this one.